app/models/photo.rb

```diff
@@ -81,12 +81,12 @@ def ownership_of_status_message
     end
   end
 
-  def self.diaspora_initialize(params = {})
-    photo = self.new params.to_hash.slice(:text, :pending)
-    photo.author = params[:author]
-    photo.public = params[:public] if params[:public]
-    photo.pending = params[:pending] if params[:pending]
+  class << self
+    alias_method :diaspora_initialize_shareable, :diaspora_initialize
+  end
```

Comment:
Well, yes, there is now `diaspora_initialize` in both `Photo` and `Post` which do the same but implemented differently. I wanted to fix this.

But this aliasing is not too pretty.

What if I make method `shareable_initialize` in Diaspora::Shareable instead and call it from `Photo` and `Post` methods `diaspora_initialize`?